services/oauth/oauth.go

```diff
@@ -406,83 +413,90 @@ func (refStats *OAuthStats) SendCountStat() {
 	}).Increment()
 }
 
-func (authErrHandler *OAuthErrResHandler) DisableDestination(destination *backendconfig.DestinationT, workspaceId, rudderAccountId string) (statusCode int, respBody string) {
+func (authErrHandler *OAuthErrResHandler) UpdateAuthStatusToInactive(destination *backendconfig.DestinationT, workspaceId, rudderAccountId string) (statusCode int, respBody string) {
 	authErrHandlerTimeStart := time.Now()
 	destinationId := destination.ID
-	disableDestMutex := authErrHandler.getKeyMutex(authErrHandler.destLockMap, destinationId)
+	authStatusInactiveMutex := authErrHandler.getKeyMutex(authErrHandler.destLockMap, destinationId)
 
-	disableDestStats := &OAuthStats{
+	getStatName := func(statName string) string {
+		return fmt.Sprintf("auth_status_inactive_%v", statName)
+	}
+
+	authStatusInactiveStats := &OAuthStats{
 		id:              destinationId,
 		workspaceId:     workspaceId,
 		rudderCategory:  "destination",
 		statName:        "",
 		isCallToCpApi:   false,
-		authErrCategory: DISABLE_DEST,
+		authErrCategory: AUTH_STATUS_INACTIVE,
 		errorMessage:    "",
 		destDefName:     destination.DestinationDefinition.Name,
 		flowType:        authErrHandler.rudderFlowType,
 	}
 	defer func() {
-		disableDestStats.statName = "disable_destination_total_req_latency"
-		disableDestStats.isCallToCpApi = false
-		disableDestStats.SendTimerStats(authErrHandlerTimeStart)
+		authStatusInactiveStats.statName = getStatName("total_req_latency")
+		authStatusInactiveStats.isCallToCpApi = false
+		authStatusInactiveStats.SendTimerStats(authErrHandlerTimeStart)
 	}()
 
-	disableDestMutex.Lock()
-	isDisableDestActive, isDisableDestReqPresent := authErrHandler.disableDestActiveMap[destinationId]
-	disableActiveReq := strconv.FormatBool(isDisableDestReqPresent && isDisableDestActive)
-	if isDisableDestReqPresent && isDisableDestActive {
-		disableDestMutex.Unlock()
-		authErrHandler.logger.Debugf("[%s request] :: Disable Destination Active : %s\n", loggerNm, disableActiveReq)
-		return http.StatusOK, fmt.Sprintf(`{response: {isDisabled: %v, activeRequest: %v}`, false, disableActiveReq)
+	authStatusInactiveMutex.Lock()
+	isAuthStatusUpdateActive, isAuthStatusUpdateReqPresent := authErrHandler.authStatusUpdateActiveMap[destinationId]
+	authStatusUpdateActiveReq := strconv.FormatBool(isAuthStatusUpdateReqPresent && isAuthStatusUpdateActive)
+	if isAuthStatusUpdateReqPresent && isAuthStatusUpdateActive {
```

Comment:
This doesn't seem to be covered. Can we test it?

Reply:
Added tests in `oauth_test.go`. Can you check and confirm ?